Testing/SystemTests/tests/analysis/PearlPowderDiffractionScriptTest.py

```diff
@@ -192,3 +192,212 @@ def test_saved_data_files(self):
+                          'PEARL/Calibration_Test/Calibration/van_spline_TT35_cycle_15_3.nxs',
+                          'PEARL/Calibration_Test/Calibration/van_spline_TT70_cycle_15_3.nxs',
+                          'PEARL/Calibration_Test/Calibration/van_spline_TT88_cycle_15_3.nxs'))
+        self._clean_up_files(filenames, DIRS)
+
+
+# ======================================================================
+# work horse
+
+class LoadCalibTests(unittest.TestCase):
+    wsname = "__LoadTest"
+    cleanup_names = []
+
+    def tearDown(self):
+        self.cleanup_names.append(self.wsname)
+        for name in self.cleanup_names:
+            try:
+                AnalysisDataService.remove(name)
+            except KeyError:
+                pass
+        self.cleanup_names = []
+
+    # ============================ Success =============================
+
+    def test_calfile_with_workspace(self):
+
+        cal_group_file = os.path.join(DIRS[0], 'PEARL/Calibration_Test/Calibration/test_cal_group_15_3.cal')
+        cal_offset_file = os.path.join(DIRS[0], 'PEARL/Calibration_Test/Calibration/pearl_offset_15_3.cal')
+        LoadCalFile(InstrumentName="PEARL", CalFilename=cal_group_file,
+                    WorkspaceName='cal_group_15_3')
+        LoadCalFile(InstrumentName="PEARL", CalFilename=cal_offset_file,
+                    WorkspaceName="cal_offset_15_3")
+
+        cal_workspaces = []
+        for i in range(0, 2):
+            cal_type = 'group'
+            if i == 1:
+                cal_type = 'offset'
+            cal_workspaces.extend((mtd['cal_' + cal_type + '_15_3_group'],
+                                   mtd['cal_' + cal_type + '_15_3_mask'],
+                                   mtd['cal_' + cal_type + '_15_3_offsets']))
+
+        for workspace in cal_workspaces:
+            self.assertTrue(isinstance(workspace, MatrixWorkspace))
+            self.assertEquals(1056, workspace.getNumberHistograms())
+            self.assertEquals(1, workspace.blocksize())
+
+        table_workspaces = []
+        table_workspaces.extend((mtd['cal_group_15_3_cal'], mtd['cal_offset_15_3_cal']))
+        for table in table_workspaces:
+            self.assertTrue(isinstance(table, ITableWorkspace))
+
+        for i in range(0, 1000, 50):
+            self.assertEqual(table_workspaces[0].cell(i, 0), table_workspaces[0].cell(i, 0))
+
+        diff_places = 2
+
+        self.assertAlmostEqual(4491.84, table_workspaces[0].cell(0, 1), places=diff_places)
+        self.assertAlmostEqual(4826.7, table_workspaces[1].cell(34, 1), places=diff_places)
+        self.assertAlmostEqual(4814.51, table_workspaces[0].cell(190, 1), places=diff_places)
+        self.assertAlmostEqual(4639.09, table_workspaces[1].cell(405, 1), places=diff_places)
+        self.assertAlmostEqual(5219.48, table_workspaces[0].cell(703, 1), places=diff_places)
+        self.assertAlmostEqual(3018.97, table_workspaces[1].cell(1055, 1), places=diff_places)
```

Comment:
For next modifications it would be good to add "reference files" and use full validate/comparison functions, as they used to do for example in the old PEARL test: https://github.com/mantidproject/mantid/blob/master/Testing/SystemTests/tests/analysis/PEARLSystemTest.py#L59.
This shouldn't be much slower, at least for the text files (GSS, Cal) and the nexus files that are not too big.